api/src/test/java/org/openmrs/api/ProviderServiceTest.java

```diff
@@ -558,4 +559,16 @@ public void getCountOfProviders_shouldExcludeRetiredProviders() throws Exception
 	public void getCountOfProviders_shouldIncludeRetiredProvidersIfIncludeRetiredIsSetToTrue() throws Exception {
 		assertEquals(4, service.getCountOfProviders("provider").intValue());
 	}
+	
+	@Test
+	public void createProviderFromUser_shouldGetUserAndCreateProvider() {
+		User u = Context.getUserService().getUser(19901);
+		
+		Assert.assertNotNull(Context.getPersonService().getPerson(19901));
+		Assert.assertNotNull(Context.getPersonService().getPersonName(19901));
+		Assert.assertNotNull(u);
```

Comment:
You need a sanity check to ensure that there is always no provider account for the user.